pkgs/dart_services/lib/src/sdk.dart

```diff
@@ -96,10 +86,41 @@ class Sdk {
     return path.join(_flutterBinPath, 'cache', 'flutter_web_sdk', 'kernel');
   }
 
-  static String _readVersionFile(String filePath) =>
-      _readFile(path.join(filePath, 'version'));
-}
+  bool get usesNewBootstrapEngine {
+    final uiWebPackage =
+        path.join(_flutterBinPath, 'cache', 'flutter_web_sdk', 'lib', 'ui_web');
+    final initializationLibrary =
+        path.join(uiWebPackage, 'ui_web', 'initialization.dart');
 
-const channels = ['stable', 'beta', 'main'];
+    final file = File(initializationLibrary);
+    if (!file.existsSync()) return false;
 
-String _readFile(String filePath) => File(filePath).readAsStringSync().trim();
+    // Look for 'Future<void> bootstrapEngine({ ... }) { ... }'.
+    return file.readAsStringSync().contains('bootstrapEngine(');
+  }
+
+  Map<String, dynamic> _callFlutterVersion() {
+    // Note that we try twice here as the 'flutter --version --machine' command
+    // can (erroneously) emit non-json text to stdout (for example, an initial
+    // analytics disclaimer).
```

Comment:
Maybe we need to run `flutter config --no-analytics` and `dart --disable-analytics` somewhere?

Reply:
`flutter config --no-analytics` is a good call. But `flutter --version --machine` - which emits json - should never emit cruft to stdout; the output needs to be parseable.